frontend/test/pytest/test_peephole_optimizations.py

```diff
@@ -23,6 +23,77 @@
 
 # pylint: disable=missing-function-docstring
 
+#
+# Complex_merging_rotations
+#
+
+# Parameterize test with different angle sets for qml.Rot and qml.CRot to ensure coverage of complex cases.
+@pytest.mark.parametrize("params1, params2", [
```

Comment:
So testing this new merge rotation pattern is a bit tricky: we know that regardless of whether the merge rotation transformation took effect or not, the circuit will produce the same results. Given that, does this test here actually test for whether the rotation gates are merged? If not, what is the best way to test that the rotation gates are merged, and what is the purpose of these end-to-end circuit execution tests?

Hint: search through the code base and look for how the existing merge rotation patterns are tested!

Reply:
Thank you @paul0403 
So, do you mean that your test for regular merging rotations (not for merging non-commutative rotations) was unsuitable and that we need a different test even for regular merging rotations?

Reply:
Thank you @paul0403 

I’ve added explicit checks in test_complex_merge_rotation to verify that the merge_rotations transformation reduces the number of rotation gates (Rot and CRot) and preserves the circuit's functionality. By explicitly calling qml.transforms.merge_rotations, we can compare the unoptimized and optimized circuits directly. This allows us to confirm both that the rotation gates are actually merged (fewer gates) and that the results remain the same, addressing the need for both functionality and transformation verification in the test. Please let me know if this method is correct.

Reply:
Thanks for adding the test! An additional approach on top of plain functionality check is definitely good to have :100: